### ISSUE TITLE
Build timeouts in Travis

### ISSUE THREAD

**Issue by adamjstewart** [spack/spack#5049]
Our build tests have been timing out at 50 minutes quite frequently lately. I've been restarting Travis jobs for around a third of new PRs. Perhaps we should build fewer packages, or remove the build tests altogether.

**Comment by alalazo** [comment]
Maybe removing tests altogether is not a good choice, as they increase coverage by quite a lot. But I definitely agree that we need to make them faster, and maybe building fewer packages is the path to take.

Looking at the current spec we build:
```console
$ spack spec -l hypre ^mpich
Input spec
--------------------------------
ofvpbtf  hypre
ty5c2nn      ^mpich

Normalized
--------------------------------
lrr2k5c  hypre
tlulk6o      ^blas
qyk5ol5      ^lapack
ty5c2nn      ^mpich

Concretized
--------------------------------
yk22xef  hypre@2.11.2%gcc@4.8~int64+internal-superlu+shared arch=linux-ubuntu14-x86_64 
od7rbtn      ^mpich@3.2%gcc@4.8+hydra+pmi+romio~verbs arch=linux-ubuntu14-x86_64 
vfbctwp      ^openblas@0.2.20%gcc@4.8~openmp+pic+shared arch=linux-ubuntu14-x86_64 
```
I guess this was chosen because it builds both a common `mpi` and a common `blas` provider (plus a package depending on them). In this way we will be ensured that `openblas` and `mpich` won't be broken, at least on the platform we test on.

I think these kind of tests belong to the CDash / build-automation support that is currently being worked. So maybe a first step in speeding up tests is to see how much coverage will decrease if we build e.g. `mpich` only?

**Comment by adamjstewart** [comment]
I agree, we could probably find a package from each build system that builds in a minute or less. That would increase coverage quite a bit.

**Comment by adamjstewart** [comment]
Closed by #5104 (also see #5068 for original work).
